sdk/src/beta9/abstractions/function.py

```diff
@@ -195,8 +204,148 @@ def _threaded_map(self, inputs: Sequence) -> Iterator[Any]:
     def map(self, inputs: Sequence[Any]) -> Iterator[Any]:
         if not self.parent.prepare_runtime(
             func=self.func,
-            stub_type=FUNCTION_STUB_TYPE,
+            stub_type=self.base_stub_type,
         ):
             terminal.error("Function failed to prepare runtime ❌")
 
         return self._threaded_map(inputs)
+
+
+class ScheduleWrapper(_CallableWrapper):
+    base_stub_type = SCHEDULE_STUB_TYPE
+    deployment_stub_type = SCHEDULE_DEPLOYMENT_STUB_TYPE
+
+    @override
+    def deploy(self, *args: List[Any], **kwargs: Dict[str, Any]) -> bool:
+        when_now = self.parent.when
+        when_utc = self.convert_schedule_to_utc(when_now)
+
+        def callback() -> None:
+            tz = time.tzname[time.localtime().tm_isdst]
+            terminal.header("Schedule details")
+            terminal.print(f"[green]Schedule ({tz}):[/green] {when_now}")
+            terminal.print(f"[green]Schedule (UTC):[/green] {when_utc}")
+
+        deployed = super().deploy(invocation_details_callback=callback, *args, **kwargs)
+        if deployed:
+            res = self.parent.function_stub.function_schedule(
+                FunctionScheduleRequest(
+                    stub_id=self.parent.stub_id,
+                    when=when_utc,
+                    deployment_id=self.deployment_id,
+                )
+            )
+            if not res.ok:
+                terminal.error(res.err_msg, exit=False)
+                return False
+        return deployed
+
+    def convert_schedule_to_utc(self, expression: str) -> str:
```

Comment:
would probably be worth adding a test for this function

Reply:
Removed the UTC conversion because if we convert the user's timezone to UTC, we're not/can't take into account whether its daylight savings or not. Its best to keep this as just UTC for everyone.